src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs

```diff
@@ -45,20 +45,32 @@ public ConsoleLoggerProvider(IConsoleLoggerSettings settings)
 
         private void OnConfigurationReload(object state)
         {
-            // The settings object needs to change here, because the old one is probably holding on
-            // to an old change token.
-            _settings = _settings.Reload();
+            try
+            {
+                // The settings object needs to change here, because the old one is probably holding on
+                // to an old change token.
+                _settings = _settings.Reload();
 
-            foreach (var logger in _loggers.Values)
+                foreach (var logger in _loggers.Values)
+                {
+                    logger.Filter = GetFilter(logger.Name, _settings);
```

Comment:
I agree. And as long as the reload token mechanic works the next time the configuration changes and eventually fixes the broken things, its all good...